cairo/ethereum/cancun/main.cairo

```diff
@@ -0,0 +1,66 @@
+%builtins output pedersen range_check ecdsa bitwise ec_op keccak poseidon range_check96 add_mod mul_mod
+// In proof mode running with RustVM requires declaring all builtins and taking them as entrypoint
+// This is probably a mis-handling from the RustVM side.
+
+from starkware.cairo.common.cairo_builtins import (
+    BitwiseBuiltin,
+    KeccakBuiltin,
+    PoseidonBuiltin,
+    ModBuiltin,
+    HashBuiltin,
+    SignatureBuiltin,
+    EcOpBuiltin,
+)
+from ethereum.cancun.fork import state_transition, BlockChain, Block
+from ethereum_types.bytes import Bytes32
+
+func main{
+    output_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr,
+    ecdsa_ptr: SignatureBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    ec_op_ptr: EcOpBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+}() {
+    alloc_locals;
+
+    local chain: BlockChain;
+    local block: Block;
+    local block_hash: Bytes32;
```

Comment:
why do we need this?

Reply:
assert that the hash of our new block matches the expected one. 

[Public inputs: pre-state root R, post-state root R' , block hash H](https://vitalik.eth.limo/general/2024/10/23/futures4.html)